File: Cards/Priest.lua
local ____lualib = require("lualib_bundle")
local __TS__Class = ____lualib.__TS__Class
local __TS__ClassExtends = ____lualib.__TS__ClassExtends
local ____exports = {}
local ____Enums = require("Enums")
local Ranks = ____Enums.Ranks
local ____Card = require("Cards.Card")
local Card = ____Card.default
____exports.default = __TS__Class()
local Priest = ____exports.default
Priest.name = "Priest"
__TS__ClassExtends(Priest, Card)
function Priest.prototype.____constructor(self, gameManager, suit)
    Card.prototype.____constructor(
        self,
        gameManager,
        suit,
        Ranks.PRIEST,
        4,
        4,
        "Priest"
    )
end
return ____exports

File: Cards/Priest.ts
import { Ranks, Suits } from "../Enums";
import GameManager from "../GameManager";
import Card from "./Card";

export default class Priest extends Card {
  constructor(gameManager: GameManager, suit: Suits) {
    super(
      gameManager,
      suit,
      Ranks.PRIEST,
      4,
      4,
      "Priest",
      "Assets/Images/PriestRank.png",
      {
        edelName: "Pope",
        edelPower: 14,
        edelValue: 4,
        edelRankAssetPath: "Assets/Images/PopeRank.png",
      }
    );
  }
}
